ui-admin/src/study/settings/SettingsView.tsx

```diff
@@ -93,6 +92,13 @@ export function LoadedSettingsView(
     }, { setIsLoading: setIsLoadingStudyConfig })
   }
 
+  useEffect(() => {
+    setPortalConfig(portalEnv.portalEnvironmentConfig)
+    setHasPortalConfigChanged(false)
+    setStudyConfig(studyEnvContext.currentEnv.studyEnvironmentConfig)
+    setHasStudyConfigChanged(false)
+  }, [studyEnvContext.currentEnv.environmentName])
```

Comment:
very nice! I think we need to do the same for `kits` and `languages` in their respective settings pages.

Reply:
An easier way to do this is to add a `key` to the place where the component is invoked.  E.g. edit line 127 of StudyEnvironmentRouter to add `key={studyEnvContext.studyEnvPath}`. That tells react to destroy and recreate the whole component if the key changes.  Then you don't have to  worry about manually resetting every bit of state.  Alternatively, if you don't want StudyEnvRouter to have to `know` when to reset the component, you can add a wrapper component around LoadedSettingsView with the key